core/src/main/java/hudson/search/FixedSet.java

```diff
@@ -51,7 +51,7 @@ public void find(String token, List<SearchItem> result) {
                 token=token.toLowerCase();
                 name=name.toLowerCase();
             }
-            if(token.equals(i.getSearchName()))
+            if(token.equals(name))
```

Comment:
It's probably quite rare case, but I created [JENKINS-42960](https://issues.jenkins-ci.org/browse/JENKINS-42960)